answerking_app/models/serializers.py

```
@@ -0,0 +1,106 @@
+import re
+
+from django.core.validators import (
+    MaxValueValidator,
+    MinValueValidator,
+    RegexValidator,
+)
+from rest_framework import serializers
+from typing_extensions import Required
+
+from answerking_app.models.models import Category, Item, Order, OrderLine
+
+MAXNUMBERSIZE = 2147483647
+
+
+def compress_white_spaces(value: str) -> str:
+    return re.sub(" +", " ", value)
+
+
+class ItemSerializer(serializers.ModelSerializer):
+    name = serializers.CharField(
+        max_length=50,
+        allow_blank=False,
+        validators=[RegexValidator("^[a-zA-Z !]+$")],
```

I'm not sure is it's needed because the model also check - but you can add unique validator
https://www.django-rest-framework.org/api-guide/validators/#uniquevalidator

---

yeah, it's also checked by the model, should be alright

---

Yeah, but maybe it's not good practise to let the model throw the error. not sure about that.

---

uh, right now it's actually hard coded and doesn't use the model, might change that